### ISSUE TITLE
could we please transfer this repository to oda-hub/oda_api?

### ISSUE THREAD

**Issue by volodymyrss** [andreatramacere/oda_api#5]

**Comment by volodymyrss** [comment]
@andreatramacere could you please?

**Comment by andreatramacere** [comment]
Hi,
we decided that I will add you as co-maintainer, but since it is forked, I think that for the time being, we can keep things as they are. When you have done changes and fixing on the forked  you can just open a pull request, that is what I do with Isabelle with Primal on Euclid. When she find something to change on his fork,  and has fixed the issue, she opens a pull request on mine.

**Comment by volodymyrss** [comment]
It's rather confusing that all repositories are on oda-hub and this one is not. 
For example, issues will be open in here and not oda-hub organization, and can not be viewed together quite as easily.
But, if you insist, ok, lets he how this will work.

**Comment by andreatramacere** [comment]
Ok, let's finish the auth issue and then fix this one. At the moment I am doing several things at the same time. I guess it is not urgent and we can do it in rather short amount of time, so we are also sure that everything has been synchronised.

**Comment by volodymyrss** [comment]
to be clear, the workflow would be precisely the same describe, of course. Since it's what everybody uses. Only, the repository would be on oda-hub organization, and PR would be made to there, giving you and me equal rights to merge them. This would implement the practice of co-maintaining.
But the point of central issue management is even more practically relevant.

So see as you wish!

**Comment by ferrigno** [comment]
So this is why, I have not the possibility to open issues on the oda_hub repository ... another reason to transfer it if you cannot maintain it.

**Comment by andreatramacere** [comment]
@ferrigno @volodymyrss Hi,
This week I am very very busy. We can discuss this next week, it this is possible for you.
Thanks
Andrea

**Comment by volodymyrss** [comment]
Absolutely! 

Here I added an item to the agenda:

https://github.com/oda-hub/meetings/blob/main/2021-02-22/agenda.md

This motion is suggested precisely because you are very busy often, and do we expect it will be different in the future?..
But let us discuss and consider.

In principle, it is not critical either way. There are many ways to operate. I will present few arguments for-against different modes, quickly.

Cheers

V.

**Comment by andreatramacere** [comment]
There is no problem for me to move the repository(ies) if:
For all the projects that I have developed from the scratch: dispatcher, MAGIC, ANTARES, oda_api, both in the README.md and in the read the docs page, my name is reported as creator and co-responsible/co-maintainer.
All these things (and further) can be fixed with an MOU that we can discuss during a dedicated meeting (not the weekly one).
Cheers,
Andrea

**Comment by volodymyrss** [comment]
Of course, we can discuss!

It remains as so. In fact, you are currently listed as sole author in most of them, even those I have been updating. But I add my name as a contribution grows.

I do think we can bring it up on the meeting and then plan further discussion.
Ok?

One immediate note, I am not sure what is the purpose of an MoU, as the furtheer is defined in the LICENSE, no? Isn't this the exact purpose of the license?

Note, also, that we are not entirely free (due to contractual obligations on the IP rights) on how to choose the licenses, have had [specific discussion with UNIGE on this topic](https://dataforum.unige.ch/t/recommended-license-for-open-data-and-software/201/2) (link restricted to UNIGE accounts) for my own selfish purposes, and got some specific policy advice. 

I could imagine if we create an MoU we would specify your continuous contribution, under some constrains on the side of the other contributors of project (taking into account approval of the management in all cases, and remaining contractual obligations). Something like that? Open for discussion! 
Maybe you have an example of such an MoU so we can review it in advance?

Clearly, this can wait till next week.

**Comment by andreatramacere** [comment]
Hi Volodymyr,
not only related to my repos, but in general, since in the future the team might change (none of us has a permanent contract), I think that we should ensure that we all keep some rights on the scientific output of the project. Of course, this concerning is due to the transitory nature of our positions. So, in the interest of everyone, it would be nice, that on top of the licence, oda will have a few rules that protect our work. I think that this  reasonable, and in the interest of all of us. Anyhow, this topic can be a bit difficult to setup and frame correctly, that's why I propose a dedicated meeting. Indeed, t, oda_hub, I guess, it is not currently under the scope of UNIGE, and there might be potential issues, since the code we develop using the university money, at some extent, belongs to the university. So, my point was to bring your attention to this point, and to find a solution that protects the original core team of oda from future unknown developments.

**Comment by ferrigno** [comment]
@andreatramacere an MOU has no meaning as we all work for UNIGE and all our work and code is intellectual property of the university. However, what you ask is more than reasonable, actually a good practice.

**Comment by andreatramacere** [comment]
@ferrigno I think that I explained what I mean with MOU in my previous post.

**Comment by volodymyrss** [comment]
Hi!

> not only related to my repos, but in general, since in the future the team might change (none of us has a permanent contract), I think that we should ensure that we all keep some rights on the scientific output of the project. Of course, this concerning is due to the transitory nature of our positions. 

Absolutely! This is precisely the reason I asked [this](https://dataforum.unige.ch/t/recommended-license-for-open-data-and-software/201/2). 
My current impression is that it is best to share all openly the permissive licenses (as permitted and indeed encouraged by UNIGE), and if/when we move-on, we can use results of our work just as we would use any other code with permissive license. 
UNIGE [confirmed](https://dataforum.unige.ch/t/recommended-license-for-open-data-and-software/201/2) this is allowed.

> So, in the interest of everyone, it would be nice, that on top of the licence, oda will have a few rules that protect our work. I think that this reasonable, and in the interest of all of us. 

An agreement, if binding, would need to be validated by UNIGE legal.

> Anyhow, this topic can be a bit difficult to setup and frame correctly, that's why I propose a dedicated meeting. 

So we agree on the date next Monday? Or  make a doodle?

> Indeed, t, oda_hub, I guess, it is not currently under the scope of UNIGE, and there might be potential issues, since the code we develop using the university money, at some extent, belongs to the university. 

I have an impression that not really to some extent, but it plainly does. And it is not up to any of us to decide on that.

> So, my point was to bring your attention to this point, and to find a solution that protects the original core team of oda from future unknown developments.

Basically technically it is implemented by having all original core team as equal owners of oda-hub.

**Comment by volodymyrss** [comment]
> @andreatramacere an MOU has no meaning as we all work for UNIGE and all our work and code is intellectual property of the university. However, what you ask is more than reasonable, actually a good practice.

Exactly! 
The precise request is quite clear and well-respected.
And it might be that it does not have to be enforced by any MoU, and in fact it might not be possible that it would be, as it would contradict UNIGE rules and related licenses.

Another problem of this being a multi-institute project could be resolved in exactly the same way - as long as all code has permissive license, and it is approved by the funding agencies (normally, it is), there is no problem, no need, and indeed no possibility for any additional agreements.

**Comment by ferrigno** [comment]
So, there is no need for further meeting as what you propose finds full support and agreement among us.
In any case, if this is accidentally neglected, you can always make a correction in the repository.

**Comment by andreatramacere** [comment]
@volodymyrss 
Indeed, since it is very complicated to enter on the UNIGE legal side,  as I stated previously, on top of the license that  has to be open, we should add a few requirements on the acknowledgements, as we always do with opensource code. Even if it has not a fully legal value, it lives in the domain of gentleman agreements, and being clearly written somewhere makes it less ambiguous. We should just write down a few lines of personal agreement, and be confident that we will deal with polite people in the future.  I think that I was quite clear to say that I don't want to go deep in legal arguments.

**Comment by andreatramacere** [comment]
@ferrigno I think that having something written on the oda_hub page would be worth.

**Comment by ferrigno** [comment]
@andreatramacere fine with me, would you propse a text and a place to show it ?

**Comment by volodymyrss** [comment]
> @volodymyrss
> Indeed, since it is very complicated to enter on the UNIGE legal side, as I stated previously, on top of the license that has to be open, we should add a few requirements on the acknowledgements, as we always do with opensource code. 

Indeed we do, but it's called attribution, and is sometimes included in licenses. Or not. But the place of this kind of constrain is generally in the license.  For example, that's large part of the purpose of CC-BY. There are options for source too.
We can start with a  code-of-conduct just as an idea (keeping in mind it is draft an up to discussion, and has limited scope):

https://github.com/oda-hub/code-of-conduct/blob/main/README.md

> Even if it has not a fully legal value, it lives in the domain of gentleman agreements, and being clearly written somewhere makes it less ambiguous. We should just write down a few lines of personal agreement, and be confident that we will deal with polite people in the future. I think that I was quite clear to say that I don't want to go deep in legal arguments.

Indeed, but why neglect function of license made for this purposes precisely?  I think MIT license [might imply that](https://softwareengineering.stackexchange.com/questions/218331/what-are-the-requirements-for-attribution-in-the-mit-license)

**Comment by volodymyrss** [comment]
> @andreatramacere fine with me, would you propse a text and a place to show it ?

turns out I already started it before seeing the suggestions! please feel free to edit.

but, I still think attribution in licenses, e.g. MIT, does just that.

I like an additional [note](https://github.com/oda-hub/code-of-conduct/blob/main/README.md) to provide a refence and answers to similar questions in the future.

**Comment by volodymyrss** [comment]
I renamed the fork, and made another repository with the same name: https://github.com/oda-hub/oda_api
All the content is identical, so it has the same effect for our purposes, I guess.
So it seems like the easiest way. It was also possible to rename and transfer, which would preserve the issues, but let's just recreate them since they are not many anyway.
